src/alexandria3k/uspto_zip_cache.py

```diff
@@ -32,15 +32,27 @@ class UsptoZipCache:
 
     def __init__(self):
         self.cached_path = None
-        self.cached_data = None
+        self.cached_data = []
         self.file_name = None
 
-    def read(self, zip_path):
-        """Return a list of XML files in the specified zip file"""
+    def read(self, zip_path, xml_sampling=lambda x: (True, True)):
+        """Return a list of XML containers in the specified zip file.
+        When the second variable of the callable returns `True` the
```

Comment:
The tuple isn't callable, its elements are. So better: "When the tuple's second element returns …"